### ISSUE TITLE
Problem  on getting the event to work with  python flask

### ISSUE THREAD

**Issue by monkyaro** [kalisio/keycloak-event-gateway#1]
I have been looking for something like this for a long time. Fortunately, when I found it, I was so excited. However, when I tried to use my endpoint for Keycloak to emit the events, I received nothing. When I checked the Keycloak logs, I found the following error:

ead-19) While sending JSON to: http://localhost:7000/test/eventListener: org.apache.http.conn.HttpHostConnectException: Connect to localhost:7000 [localhost/127.0.0.1] failed: Connection refused

But when I test my endpoint on Postman, it is working. Please help me understand what is wrong with my configuration.   iam using flask

**Comment by dandriana** [comment]
Hello, can Keycloak see your Flask app via “localhost”, or do you need to give it a full IP address?

**Comment by monkyaro** [comment]
@dandriana Yes, I am using the same Keycloak to authenticate and authorize users. However, when it comes to emitting events, I encounter the error mentioned above.

**Comment by dandriana** [comment]
How are your Keycloak instance and your Flask app deployed? In Docker containers? What is the command line by which you start them?

**Comment by monkyaro** [comment]
Yes, both are running in Docker containers, but I deployed them using different Docker Compose files. For Keycloak, I used version 22.05 and the keycloak-event-gateway.jar. I followed the instructions you provided on how to download it from Docker Hub. I did not use Maven because I am not very familiar with Java.

**Comment by dandriana** [comment]
If they are running in different Docker containers, it may be that “localhost” doesn’t mean for Keycloak the same as for your browser.
You have two alternatives:

1. Give Keycloak the real Flask’s IP Address (e.g. “http://172.17.0.2:7000/test/eventListener” instead of “http://localhost:7000/test/eventListener”, check your container’s actual IP Address, usually starting with 172.17.0.)
2. Start Keycloak with the ”network_mode: host” option in docker-compose.yml. It tells Docker to give Keycloak container the same network access as its host. In that case, “http://localhost:7000/test/eventListener” should work.

**Comment by monkyaro** [comment]
I apologize for the confusion. They are actually running in the same Docker container, but Flask and Keycloak are initiated from different Docker Compose files

**Comment by dandriana** [comment]
I guess two different docker-compose files mean two different containers. What does “docker ps” tell you?

**Comment by monkyaro** [comment]
I attempted to use a non-existing endpoint, but I received the same results as with the existing one:
For the non-existing endpoint:
"http://localhost:7000/test/eventListener1: org.apache.http.conn.HttpHostConnectException: Connect to localhost:7000 [localhost/127.0.0.1] failed: Connection refused"
For the existing endpoint:
"http://localhost:7000/test/eventListener: org.apache.http.conn.HttpHostConnectException: Connect to localhost:7000 [localhost/127.0.0.1] failed: Connection refused"

**Comment by dandriana** [comment]
It is probable the problem here comes from the “localhost” keyword.

**Comment by dandriana** [comment]
I think you should inspect both your containers with the “docker inspect \<containerName\>” command and see what their IP Addresses are. Then, if they are on the same network, just use “http://\<ipAddress\>:7000/test/eventListener” to make Keycloak reach your Flask app.

**Comment by monkyaro** [comment]
Thank you very much for your assistance. It's now working. I realized I was confusing the localhost of the host machine with the localhost of the Docker container. I managed to resolve the issue by following your instructions.
